zilliqa/src/pool.rs

```diff
@@ -192,6 +192,21 @@ impl TransactionPool {
         std::mem::take(&mut self.transactions).into_values()
     }
 
+    /// Calculate the effective nonce of the sender, if all the currently pending transactions in the pool were mined
+    /// in order. Effectively, this calculates the current nonce plus the number of transactions with sequential nonces
+    /// in the pool.
+    pub fn pending_nonce(&self, sender: Address, mut current_nonce: u64) -> u64 {
+        loop {
+            let index = TxIndex::Nonced(sender, current_nonce);
```

Comment:
@theo-zil I could do with a double check of this logic from you. We basically ignore any nonce-less transactions in the mempool. Does that seem correct?

(by the way, the use case for this API is that Eth clients call it between each transaction to set the nonce of the next one)

Reply:
Yup, looks good to me.
I mean, the name of the RPC is wrong for Zilliqa, since what it really returns is the pending nonce and not the transaction count; on eth they're the same but we break that assumption. But since eth software mostly cares about the nonce, we have to follow that usecase even though it doesn't match the name.